### PR TITLE
Port the authz example to use the Python SDK

### PR DESCRIPTION
**Pull request by Julian**
### Description 

_What behavior does this PR change, and why?_

This changes the requests-based authz example to use the [in development SDK](https://github.com/TheNileDev/nile-py).

_Note any breaking changes_

It essentially behaves the same as before -- the only real difference is that given the SDK has some Python objects (instead of dicts) for things like `Policy` objects, they now appear in output. Obviously that can be changed to just show names, or whatever.

### Submitter Validation

_Describe the validation already done, or needs to be done, by the PR submitter._

I've ran this against production already and confirmed it seems to work as before.

### Reviewer Tasks

_Describe the tasks/validation that the PR submitter is requesting to be done by the reviewer._

Obviously let me know if there's anything you want me to tweak or explain here -- I'll be writing some more basic info for the SDK in its README in the next day or two, but hopefully the standalone instructions here mean you at least are able to run it. Let me know if not!

### CODE REVIEW

File: authz-python/requirements.txt [hunk 1]
```diff
@@ -1,6 +1,41 @@
...
-#os
-#json
-#sys
-python-dotenv
-emoji
+#
+# This file is autogenerated by pip-compile with python 3.10
+# To update, run:
+#
+#    pip-compile requirements.in
```
Comment (Julian) [inline authz-python/requirements.txt:5]: As I mentioned in the commit, this is a somewhat common pattern for tracking lockfiles in Python, but means you will need to run this command to update this file whenever you want to bump its dependencies (e.g. if a new version of the SDK is released and you want the examples to use it!)

File: authz-python/authz.py [hunk 2]
```diff
@@ -1,144 +1,148 @@
...
+    print(f"{BAD} could not find {users_path}")
+    sys.exit(1)
+else:
+    # Load first user only
+    user, *_ = json.loads(contents)
+
+NILE_TENANT1_EMAIL = user["email"]
+NILE_TENANT_PASSWORD = user["password"]
+
+token = login_developer.sync(
```
Comment (Julian) [inline authz-python/authz.py:66]: (`sync` here means a synchronous API call from the Python SDK, as opposed to an async one)

File: authz-python/README.md [hunk 3]
```diff
@@ -34,12 +34,12 @@ This example uses Python and the NILE REST API to demonstrate authorization feat
 
 ## Install Dependencies
 
-Note that this example has been validated with Python 3.8.9 so please ensure you are using a compatible version.
+You'll need a version of Python at least as new as 3.7.
```
Comment (ybyzek) [inline authz-python/README.md:37]:
```suggestion
Verify you are running Python version 3.7 or newer (check with `python --version`)
```

File: authz-python/requirements.txt (hunk 1 above)
Comment (ybyzek) [reply authz-python/requirements.txt:5]: @Julian thanks for noting this.

**Review by ybyzek — COMMENTED** [review]
@Julian the example runs perfectly in my environment, thanks!

I left some minor comments, plus to note that the top-level README needs an edit:

```
- [Authorization with Python](authz-python/) | Python [-REST-]{+SDK+} | Use ABAC to grant and revoke permissions to resources
```

File: authz-python/README.md [hunk 4]
```diff
@@ -34,12 +34,12 @@ This example uses Python and the NILE REST API to demonstrate authorization feat
...
 
 Run the following command:
 
 ```
-pip3 install -r requirements.txt
+python3 -m venv venv && venv/bin/python3 install -r requirements.txt
```
[the remark this answers is not in the document]
Comment (Julian) [reply authz-python/README.md]: Thanks, fixed!

**Comment by Julian** [comment]
Thanks for the review, addressed I believe! Glad to hear it works for you as well certainly.

**Review by ybyzek — APPROVED** [review]
LGTM, thanks for the PR @Julian

**Comment by ybyzek** [comment]
@Julian are you able to merge or are you waiting on me to merge?

**Comment by Julian** [comment]
I don't have merge rights in this repo I think, I was assuming you were!

**Pull request merged**